arangod/Replication2/StateMachines/Document/DocumentFollowerState.h

```diff
@@ -81,6 +81,7 @@ struct DocumentFollowerState
   std::unique_ptr<IDocumentStateTransactionHandler> _transactionHandler;
   Guarded<GuardedData, basics::UnshackledMutex> _guardedData;
   ActiveTransactionsQueue _activeTransactions;
+  std::atomic<std::uint64_t> _snapshotsCount;
```

Comment:
It looks like _snapshotsCount is only access inside `_guardedData.doUnderLock`, so I think we can move it into GuardedData and make it non-atomic.